### ISSUE TITLE
Unable to discern a WSGI application from 'None' ???

### ISSUE THREAD

**Issue by waterbug** [hangarunderground/hendrix#89]
Seems strange to throw up your hands at that point, when it's as simple as
from django.core.wsgi import get_wsgi_application
application = get_wsgi_application()

**Comment by jMyles** [comment]
What's the scenario you are imagining?  One where you simply don't specify a WSGI application on the command line?

**Comment by waterbug** [comment]
Exactly.  The code I cited creates a wsgi application, but of course
hendrix's code may have to be slightly different, I don't know.
One thing I *do* know is that once Django has the settings and
the urls.py, it can create a wsgi application without any
further information ... so why would you provide anything on
the command line?  It would violate DRY.

**Comment by jMyles** [comment]
I think that for convenience sake, I prepared to agree with you.

However, I think it's an anti-pattern in general to provide "settings" (ie, a module formatted so as to be consumed by the LazySettings object) and have those settings go and get a WSGI application.  Most WSGI servers take a WSGI application, and PEP 0333 describes a framework and a server communicating through an application object to be specified, not the result of a settings quasi-class or a framework-side method that fetches the proper object.

An even stronger argument is: How do we know for sure that the user intends to deploy django?  What if they have django installed in the environ but are using a different WSGI application, inherited from Pyramid or Flask or elsewhere?

However, again, for convenience sake, I agree that, in the limited case that they didn't specify a WSGI application object *and* where we can use common sense to determine that they are using django and have a django WSGI application that is accessible via get_wsgi_application(), let's just use that.

Do you care to create an issue describing the exact behavior you desire?

Thanks for bringing this to our attention - we sometimes need help refraining from being purists.  :-)

**Comment by jMyles** [comment]
Also, FYI, merging #88 is probably a pre-requisite, because the parts of the code that actually ascertain the presence or absence of a WSGI application have been moved.  So, if you are inclined to work on it, I suggest you work against this fork: https://github.com/SlashRoot/hendrix

**Comment by waterbug** [comment]
Sorry, I wasn't offering to work on Hendrix, just filing a bug.  ;)

**Comment by waterbug** [comment]
Actually I don't think this is a bug, I just submitted the issue
because I couldn't find a hendrix mailing list -- sending it as a
comment to a mailing list would have been better.  Really I was
just asking because the importing of django stuff made it appear that
hendrix was to some extent aimed at django wsgi apps and if that were
the case (which actually could be made explicit, perhaps by having
a '--django' argument or something ;), then a certain logic could
be followed to provide a more lux user experience.  ;)

In my particular case, I was trying out hendrix on a django app
that is set up to use apache / mod_wsgi, so it has a little
subdirectory with the relevant apache conf sections and an
absolutely minimal "wsgi.py" that is completely generic and
has no code related to my app at all except the system paths --
it is really just the get_wsgi_application() ...
mod_wsgi (unlike hendrix) does not do any heuristics
to determine whether the wsgi app is actually a django app,
so of course in that context the developer has to provide
the "wsgi app".

**Comment by jMyles** [comment]
Understood. 

We do have a mailing list: https://groups.google.com/forum/#!forum/hendrix-dev

I do think that I prefer to 'fix' this.  @KPrasch and I were just talking the other day about how hendrix can distinguish itself a little bit from Twisted by providing "niceties."  I love that Twisted is very literal and smooth, but I also think that a web server can afford to feel more plush.
